File: backend/fleet_app/enums.py
"""后端共享枚举/常量（前后端枚举保持一致，见 README「枚举位置」）。"""


class VehicleStatus:
    AVAILABLE = 'Available'
    ON_TRIP = 'OnTrip'
    MAINTENANCE = 'Maintenance'
    RETIRED = 'Retired'
    CHOICES = [
        (AVAILABLE, '可用'),
        (ON_TRIP, '运输中'),
        (MAINTENANCE, '维保中'),
        (RETIRED, '已报废'),
    ]


class DriverStatus:
    AVAILABLE = 'Available'
    ON_TRIP = 'OnTrip'
    LEAVE = 'Leave'
    SUSPENDED = 'Suspended'
    CHOICES = [
        (AVAILABLE, '可用'),
        (ON_TRIP, '运输中'),
        (LEAVE, '休假'),
        (SUSPENDED, '停用'),
    ]


class DispatchStatus:
    PENDING = 'Pending'
    ASSIGNED = 'Assigned'
    IN_PROGRESS = 'InProgress'
    COMPLETED = 'Completed'
    CANCELLED = 'Cancelled'

    # 未结束：占用车辆与司机，禁止再被分配
    ACTIVE_STATUSES = (ASSIGNED, IN_PROGRESS)
    # 终态
    FINISHED_STATUSES = (COMPLETED, CANCELLED)


class MaintenanceType:
    ROUTINE = 'Routine'
    REPAIR = 'Repair'
    EMERGENCY = 'Emergency'
    INSPECTION = 'Inspection'


# 调度单状态流转白名单：当前状态 -> 允许进入的状态
DISPATCH_TRANSITIONS = {
    DispatchStatus.ASSIGNED: (DispatchStatus.IN_PROGRESS, DispatchStatus.CANCELLED),
    DispatchStatus.IN_PROGRESS: (DispatchStatus.COMPLETED, DispatchStatus.CANCELLED),
}

File: frontend/src/types/enums.ts
export enum VehicleStatus { Available = 'Available', OnTrip = 'OnTrip', Maintenance = 'Maintenance', Retired = 'Retired' }
export enum DispatchStatus { Pending = 'Pending', Assigned = 'Assigned', InProgress = 'InProgress', Completed = 'Completed', Cancelled = 'Cancelled' }
export enum MaintenanceType { Routine = 'Routine', Repair = 'Repair', Emergency = 'Emergency', Inspection = 'Inspection' }
export enum DriverStatus { Available = 'Available', OnTrip = 'OnTrip', Leave = 'Leave', Suspended = 'Suspended' }
export enum PaymentMethod { Cash = 'Cash', Card = 'Card', Company = 'Company' }

// 后端业务错误码（与 backend/fleet_app/exceptions.py、dispatch_service.py 对应）
export enum DispatchErrorCode {
  VehicleUnavailable = 'vehicle_unavailable',
  VehicleInsuranceExpired = 'vehicle_insurance_expired',
  LoadExceeded = 'load_exceeded',
  DriverUnavailable = 'driver_unavailable',
  DriverLicenseExpired = 'driver_license_expired',
  VehicleBusy = 'vehicle_busy',
  DriverBusy = 'driver_busy',
  IllegalTransition = 'illegal_transition',
  DuplicateOperation = 'duplicate_operation',
  AlreadyFinished = 'already_finished',
  NotFound = 'not_found',
}
